launcher/ui/pages/modplatform/ResourcePage.cpp

```diff
@@ -354,14 +355,22 @@ void ResourcePage::removeResourceFromPage(const QString& name)
 
 void ResourcePage::onResourceSelected()
 {
-    if (m_selected_version_index < 0)
+    if (m_selectedVersionIndex < 0)
         return;
 
     auto current_pack = getCurrentPack();
-    if (!current_pack || !current_pack->versionsLoaded)
+    if (!current_pack || !current_pack->versionsLoaded || current_pack->versions.size() < m_selectedVersionIndex)
         return;
 
-    auto& version = current_pack->versions[m_selected_version_index];
+    auto& version = current_pack->versions[m_selectedVersionIndex];
+    if (version.downloadUrl.isNull()) {
+        CustomMessageBox::selectable(this, tr("How?"),
+                                     "You managed to select a resource that doesn't have a download link. Because of this missing "
+                                     "information prism will not try to download it.",
+                                     QMessageBox::Warning)
+            ->show();
```

Comment:
Why "You managed to" and why not translated?

Reply:
Translation fine but I do not have a better text to announce the user that something went wrong

Reply:
I also updated the language.